src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/CustomSideNameTests.cs

```diff
@@ -0,0 +1,225 @@
+  }
+}
+";
+
+            ApiComparer differ = new();
+            bool enableNullable = false;
+            IAssemblySymbol left = SymbolFactory.GetAssemblyFromSyntax(leftSyntax, enableNullable);
+            IAssemblySymbol right = SymbolFactory.GetAssemblyFromSyntax(rightSyntax, enableNullable);
+            string expectedLeftName = "ref/net6.0/a.dll";
+            string expectedRightName = "lib/net6.0/a.dll";
+            differ.StrictMode = true;
+            IEnumerable<CompatDifference> differences = differ.GetDifferences(new[] { left }, new[] { right }, leftName: expectedLeftName, rightName: expectedRightName);
+            Assert.Single(differences);
+            AssertNames(differences.First(), expectedLeftName, expectedRightName, leftFirst: false);
+        }
+
+        [Fact]
+        public void MultipleRightsMetadataInformationIsUsedAsName()
+        {
+            string leftSyntax = @"
+namespace CompatTests
+{
+  public class First
+  {
+    public class FirstNested
+    {
+      public class SecondNested
+      {
+        public class ThirdNested
+        {
+          public string MyField;
+        }
+      }
+    }
+  }
+}
+";
+
+            string[] rightSyntaxes = new[]
+            { @"
+namespace CompatTests
+{
+  public class First
+  {
+    public class FirstNested
+    {
+      public class SecondNested
+      {
+        public class ThirdNested
+        {
+        }
+      }
+    }
+  }
+}
+",
+            @"
+namespace CompatTests
+{
+  public class First
+  {
+    public class FirstNested
+    {
+      public class SecondNested
+      {
+        public class ThirdNested
+        {
+        }
+      }
+    }
+  }
+}
+",
+            @"
+namespace CompatTests
+{
+  public class First
+  {
+    public class FirstNested
+    {
+      public class SecondNested
+      {
+        public class ThirdNested
+        {
+        }
+      }
+    }
+  }
+}
+"};
+
+            ApiComparer differ = new();
+            ElementContainer<IAssemblySymbol> left =
+                new(SymbolFactory.GetAssemblyFromSyntax(leftSyntax), new MetadataInformation(string.Empty, string.Empty, "ref/net6.0/a.dll"));
+
+            IList<ElementContainer<IAssemblySymbol>> right = SymbolFactory.GetElementContainersFromSyntaxes(rightSyntaxes);
+
+            IEnumerable<(MetadataInformation, MetadataInformation, IEnumerable<CompatDifference>)> differences =
+                differ.GetDifferences(left, right);
+
+            int i = 0;
+            foreach ((MetadataInformation, MetadataInformation, IEnumerable<CompatDifference> differences) diff in differences)
+            {
+                Assert.Single(diff.differences);
+                AssertNames(diff.differences.First(), left.MetadataInformation.AssemblyId, right[i++].MetadataInformation.AssemblyId);
+            }
+        }
+
+        private void AssertNames(CompatDifference difference, string expectedLeftName, string expectedRightName, bool leftFirst = true)
```

Comment:
would it make sense for compatDifference to have  left and right assembly names as fields ?

Reply:
I thought about that, and we might need it if we use the suppression engine at API Compat level, but without that, I don't think it is very relevant as that would just be information that we duplicate across all compat differences for the same tree. 

I think I would rather not store it there for the time being as the consumer already know what the values of left and right are for all the compat differences they get are.